geo/src/arrow.rs

```diff
@@ -0,0 +1,295 @@
+use std::sync::Arc;
+
+use arrow_buffer::ScalarBuffer;
+use arrow_schema::{DataType, Field};
+use geoarrow::ArrayBase;
+use geoarrow::array::{CoordBuffer, PointArray, SeparatedCoordBuffer};
+use vortex::arcref::ArcRef;
+use vortex::arrays::ExtensionArray;
+use vortex::arrow::ArrowArray;
+use vortex::arrow::compute::{ToArrowArgs, ToArrowKernelRef};
+use vortex::compute::{InvocationArgs, Kernel, Output};
+use vortex::dtype::arrow::DTypeConversion;
+use vortex::dtype::{DType, ExtDType};
+use vortex::error::{VortexResult, vortex_bail};
+use vortex::{Array, ToCanonical, register_kernel};
+
+use crate::array::GeometryArray;
+use crate::{Dimension, GeoMetadata, GeometryType, OwnedGeometryType};
+
+/// Kernel that allows converting a Vortex extension array with geometry type into a GeoArrow
+/// array layout.
+#[derive(Debug)]
+pub struct ToGeoArrow;
+
+impl Kernel for ToGeoArrow {
+    fn invoke(&self, args: &InvocationArgs) -> VortexResult<Option<Output>> {
+        let ToArrowArgs { array, .. } = ToArrowArgs::try_from(args)?;
+        match array.as_any().downcast_ref::<ExtensionArray>() {
+            None => Ok(None),
+            Some(ext_array) => {
+                if let Ok(geometry_array) = GeometryArray::try_from(ext_array) {
+                    // based on the particular geometry, encode into GeoArrow type.
+                    let array = match geometry_array {
+                        GeometryArray::Point(point_array, metadata) => {
+                            let coords = coordinate_buffer(point_array, metadata)?;
+                            let nulls = point_array.validity_mask()?.to_null_buffer();
+
+                            PointArray::new(
+                                coords,
+                                nulls,
+                                // TODO(aduffy): include CRS information from metadata
+                                Arc::default(),
+                            )
+                            .into_array_ref()
+                        }
+                        GeometryArray::LineString(array, metadata) => todo!(),
+                        GeometryArray::Polygon(array, metadata) => todo!(),
+                        GeometryArray::WKB(array, metadata) => todo!(),
+                    };
+
+                    Ok(Output::Array(ArrowArray::new(array, array.is_nullable())))
```

Comment:
Ah very neat, we can just impl `to_arrow` in this unrelated crate for conversion 👍